Spigot-Server-Patches/0084-Configurable-Chunk-Inhabited-Time.patch

```diff
@@ -0,0 +1,47 @@
+From 81880e25f1832a08a1f1253fd50b60e2fb5de71c Mon Sep 17 00:00:00 2001
+From: Aikar <aikar@aikar.co>
+Date: Mon, 28 Mar 2016 20:46:14 -0400
+Subject: [PATCH] Configurable Chunk Inhabited Time
+
+Vanilla stores how long a chunk has been active on a server, and dynamically scales some
+aspects of vanilla gameplay to this factor.
+
+For people who want all chunks to be treated equally, you can chose a fixed value.
+
+This allows to fine-tune vanilla gameplay.
+
+diff --git a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+index deb4ec25..d2559f7c 100644
+--- a/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
++++ b/src/main/java/com/destroystokyo/paper/PaperWorldConfig.java
+@@ -233,4 +233,14 @@ public class PaperWorldConfig {
+     private void firePhysicsEventForRedstone() {
+         firePhysicsEventForRedstone = getBoolean("fire-physics-event-for-redstone", firePhysicsEventForRedstone);
+     }
++
++    public int fixedInhabitedTime;
++    private void fixedInhabitedTime() {
++        if (PaperConfig.version < 16) {
++            if (!config.getBoolean("world-settings.default.use-chunk-inhabited-timer", true)) config.set("world-settings.default.fixed-chunk-inhabited-time", 0);
```

Comment:
I assume you did this from the global PaperConfig to workaround that annoying bug with `set("path", null)`?

Reply:
I did this from the global PaperConfig in order to be able to do this separately for the default world settings and the settings of the specific world.

Otherwise if a server has a world with world specific settings and other worlds using the default world settings the following would happen:
The fixedInhabitedTime() method gets executed for the world with world specific settings. In order to migrate the old option the set() method in the same class is executed. This method not only updates the value in the world specific settings but also in the default world settings. That means that the previous value in the default world settings is ignored and all other world using the default world settings will see the wrong value.